public/src/components/calls/calls_page_parent.jsx

```diff
@@ -5,43 +5,85 @@ import CallsPageMain from './calls_page_main';
 
 export default class CallPage extends Component {
   componentDidMount() {
-    console.log('#############meow', this.props);
-    // const { current_campaign,
-    //         has_user_joined_campaign,
-    //         calls_made,
-    //         next_call,
-    //         current_call,
-    //         script_questions,
-    //         current_script } = this.props;
+    const { id: campaign_id } = this.props.match.params;
+    const { user_id,
+            current_campaign,
+            fetchScript,
+            fetchScriptQuestions,
+            assignToCall } = this.props;
+    const { script_id } = current_campaign;
+    assignToCall(user_id, campaign_id);
+    fetchScript(script_id);
+    fetchScriptQuestions(script_id);
   }
 
   render() {
     const { current_campaign,
             has_user_joined_campaign,
             calls_made,
-            next_call,
             current_call,
             script_questions,
-            current_script } = this.props;
-    return (
-      <Row className="show-grid">
-        <Col xs={5} md={4} lg={3}>
-          <CallsSideBar
-            current_campaign={current_campaign}
-            joined_campaign={has_user_joined_campaign}
-            calls_made={calls_made}
-          />
-        </Col>
-        <Col xs={7} md={8} lg={9}>
-          <CallsPageMain
-            current_campaign={current_campaign}
-            current_script={current_script}
-            script_questions={script_questions}
-            current_call={current_call}
-            next_call={next_call}
-          />
-        </Col>
-      </Row>
-    );
+            current_script,
+            updateCallOutcome,
+            updateCallStatus,
+            contact_name,
+            contact_number,
+            getCallContactInfo,
+            assignToCall,
+            user_id,
+            history,
+            setCurrentCallActive,
+            setCurrentCallInactive,
+            ...storeProps } = this.props;
+    const { id } = current_campaign;
+    if (current_call) {
+      const { status,
+              outcome,
+              contact_id,
+              user_call_sid,
+              call_id,
+              call_active } = this.props;
+      return (
+        <Row className="show-grid">
+          <Col xs={5} md={4} lg={3}>
+            <CallsSideBar
+              current_campaign={current_campaign}
```

Comment:
I'm very curious what everyone thinks about this long list of props...

Reply:
As suggested above, using a current_call object to capture a bunch of these props will help. Later we can think about packaging methods in a similar way if needed.

Reply:
It actually made it more difficult to contionally render components as rerenders only occur if state changes in the store. I had an object of all of the state previously, but it ended up creating duplicate state info that got stale.